code/modules/research/xenobiology/crossbreeding/_structures.dm

```diff
@@ -0,0 +1,667 @@
+GLOBAL_LIST_EMPTY(bluespace_slime_crystals)
+
+/obj/structure/slime_crystal
+	name = "slimic pylon"
+	desc = "Glassy, pure, transparent. Powerful artifact that relays the slimecore's influence onto space around it."
+	max_integrity = 5
+	anchored = TRUE
+	density = TRUE
+	icon = 'icons/obj/slimecrossing.dmi'
+	icon_state = "slime_pylon"
+	resistance_flags = FIRE_PROOF | ACID_PROOF
+	///Assoc list of affected mobs, the key is the mob while the value of the map is the amount of ticks spent inside of the zone.
+	var/list/affected_mobs = list()
+	///Used to determine wether we use view or range
+	var/range_type = "range"
+	///What color is it?
+	var/colour
+
+/obj/structure/slime_crystal/New(loc,obj/structure/slime_crystal/master_crystal, ...)
+	. = ..()
+	if(master_crystal)
+		invisibility = INVISIBILITY_MAXIMUM
+
+/obj/structure/slime_crystal/Initialize()
+	. = ..()
+	name =  colour + " slimic pylon"
+	var/itemcolor = "#FFFFFF"
+	switch(colour)
+		if("orange")
+			itemcolor = "#FFA500"
+		if("purple")
+			itemcolor = "#B19CD9"
+		if("blue")
+			itemcolor = "#ADD8E6"
+		if("metal")
+			itemcolor = "#7E7E7E"
+		if("yellow")
+			itemcolor = "#FFFF00"
+		if("dark purple")
+			itemcolor = "#551A8B"
+		if("dark blue")
+			itemcolor = "#0000FF"
+		if("silver")
+			itemcolor = "#D3D3D3"
+		if("bluespace")
+			itemcolor = "#32CD32"
+		if("sepia")
+			itemcolor = "#704214"
+		if("cerulean")
+			itemcolor = "#2956B2"
+		if("pyrite")
+			itemcolor = "#FAFAD2"
+		if("red")
+			itemcolor = "#FF0000"
+		if("green")
+			itemcolor = "#00FF00"
+		if("pink")
+			itemcolor = "#FF69B4"
+		if("gold")
+			itemcolor = "#FFD700"
+		if("oil")
+			itemcolor = "#505050"
+		if("black")
+			itemcolor = "#000000"
+		if("light pink")
+			itemcolor = "#FFB6C1"
+		if("adamantine")
+			itemcolor = "#008B8B"
```

Comment:
Could be an idea to have an assoc list so you can just do itemcolor = colour_list[colour]